### ISSUE TITLE
XR 'openconfig acl' service trying incorrectly to change device configuration

### ISSUE THREAD

**Issue by ygorelik** [model-driven-devops/nso-oc-services#130]
The translated 'openconfig acl' service is trying incorrectly to change device configuration:
```
 devices {
     device xr {
         config {
             ipv4 {
                 access-list {
                     named-acl BFD_ECHO {
                         rule 10 {
-                            line "permit udp any any eq 3785";
+                            line "permit udp any any eq 3785 ";
                         }
                     }
                     named-acl NTP {
                         rule 10 {
-                            line "permit ipv4 host 144.133.188.239 any";
+                            line "permit ipv4 144.133.188.239 0.0.0.0 any ";
                         }
                         rule 20 {
-                            line "permit ipv4 host 144.133.188.238 any";
+                            line "permit ipv4 144.133.188.238 0.0.0.0 any ";
                         }
                         rule 30 {
-                            line "deny ipv4 any any log";
+                            line "deny ipv4 any any ";
                         }
                     }
                     named-acl SSM_RANGE {
                         rule 10 {
-                            line "permit ipv4 238.0.0.0 0.255.255.255 any";
+                            line "permit ipv4 238.0.0.0 0.255.255.255 any ";
                         }
                     }
                 }
             }
         }
```
Corresponding oc-service configuration:
```
{
  "mdd:openconfig": {
    "openconfig-acl:acl": {
      "openconfig-acl:acl-sets": {
        "openconfig-acl:acl-set": [
          {
            "openconfig-acl:name": "BFD_ECHO",
            "openconfig-acl:type": "ACL_IPV4",
            "openconfig-acl:config": {
              "openconfig-acl:name": "BFD_ECHO",
              "openconfig-acl:type": "ACL_IPV4",
              "openconfig-acl:description": "BFD_ECHO"
            },
            "openconfig-acl:acl-entries": {
              "openconfig-acl:acl-entry": [
                {
                  "openconfig-acl:sequence-id": 10,
                  "openconfig-acl:config": {
                    "openconfig-acl:sequence-id": 10
                  },
                  "openconfig-acl:actions": {
                    "openconfig-acl:config": {
                      "openconfig-acl:forwarding-action": "ACCEPT",
                      "openconfig-acl:log-action": "LOG_NONE"
                    }
                  },
                  "openconfig-acl:ipv4": {
                    "openconfig-acl:config": {
                      "openconfig-acl:protocol": "IP_UDP",
                      "openconfig-acl:source-address": "0.0.0.0/0",
                      "openconfig-acl:destination-address": "0.0.0.0/0"
                    }
                  },
                  "openconfig-acl:transport": {
                    "openconfig-acl:config": {
                      "openconfig-acl:source-port": "ANY",
                      "openconfig-acl:destination-port": 3785
                    }
                  }
                }
              ]
            }
          },
          {
            "openconfig-acl:name": "NTP",
            "openconfig-acl:type": "ACL_IPV4",
            "openconfig-acl:config": {
              "openconfig-acl:name": "NTP",
              "openconfig-acl:type": "ACL_IPV4",
              "openconfig-acl:description": "NTP"
            },
            "openconfig-acl:acl-entries": {
              "openconfig-acl:acl-entry": [
                {
                  "openconfig-acl:sequence-id": 10,
                  "openconfig-acl:config": {
                    "openconfig-acl:sequence-id": 10
                  },
                  "openconfig-acl:actions": {
                    "openconfig-acl:config": {
                      "openconfig-acl:forwarding-action": "ACCEPT",
                      "openconfig-acl:log-action": "LOG_NONE"
                    }
                  },
                  "openconfig-acl:ipv4": {
                    "openconfig-acl:config": {
                      "openconfig-acl:source-address": "144.133.188.239/32",
                      "openconfig-acl:destination-address": "0.0.0.0/0"
                    }
                  }
                },
                {
                  "openconfig-acl:sequence-id": 20,
                  "openconfig-acl:config": {
                    "openconfig-acl:sequence-id": 20
                  },
                  "openconfig-acl:actions": {
                    "openconfig-acl:config": {
                      "openconfig-acl:forwarding-action": "ACCEPT",
                      "openconfig-acl:log-action": "LOG_NONE"
                    }
                  },
                  "openconfig-acl:ipv4": {
                    "openconfig-acl:config": {
                      "openconfig-acl:source-address": "144.133.188.238/32",
                      "openconfig-acl:destination-address": "0.0.0.0/0"
                    }
                  }
                },
                {
                  "openconfig-acl:sequence-id": 30,
                  "openconfig-acl:config": {
                    "openconfig-acl:sequence-id": 30
                  },
                  "openconfig-acl:actions": {
                    "openconfig-acl:config": {
                      "openconfig-acl:forwarding-action": "REJECT",
                      "openconfig-acl:log-action": "LOG_NONE"
                    }
                  },
                  "openconfig-acl:ipv4": {
                    "openconfig-acl:config": {
                      "openconfig-acl:source-address": "0.0.0.0/0",
                      "openconfig-acl:destination-address": "0.0.0.0/0"
                    }
                  }
                }
              ]
            }
          },
          {
            "openconfig-acl:name": "SSM_RANGE",
            "openconfig-acl:type": "ACL_IPV4",
            "openconfig-acl:config": {
              "openconfig-acl:name": "SSM_RANGE",
              "openconfig-acl:type": "ACL_IPV4",
              "openconfig-acl:description": "SSM_RANGE"
            },
            "openconfig-acl:acl-entries": {
              "openconfig-acl:acl-entry": [
                {
                  "openconfig-acl:sequence-id": 10,
                  "openconfig-acl:config": {
                    "openconfig-acl:sequence-id": 10
                  },
                  "openconfig-acl:actions": {
                    "openconfig-acl:config": {
                      "openconfig-acl:forwarding-action": "ACCEPT",
                      "openconfig-acl:log-action": "LOG_NONE"
                    }
                  },
                  "openconfig-acl:ipv4": {
                    "openconfig-acl:config": {
                      "openconfig-acl:source-address": "238.0.0.0/8",
                      "openconfig-acl:destination-address": "0.0.0.0/0"
                    }
                  }
                }
              ]
            }
          }
        ]
      },
      "openconfig-acl:interfaces": {
        "openconfig-acl:interface": []
      },
      "openconfig-acl-ext:ntp": {
        "openconfig-acl-ext:peer": {
          "openconfig-acl-ext:config": {
            "openconfig-acl-ext:peer-acl-set": "NTP"
          }
        }
      }
    }
  }
}
```
The issues seems to appear in the mdd/translation code.

Device configuration attached.
[xr_config.xml.txt](https://github.com/model-driven-devops/nso-oc-services/files/10247845/xr_config.xml.txt)

**Comment by ygorelik** [comment]
After the fix I am still have one discrepancy:
```
 devices {
     device xr {
         config {
             ipv4 {
                 access-list {
                     named-acl NTP {
                         rule 30 {
-                            line "deny ipv4 any any log";
+                            line "deny ipv4 any any";
                         }
                     }
                 }
             }
         }
```

**Comment by stmosher-dev** [comment]
will check the logging issue

**Comment by stmosher-dev** [comment]
OpenConfig ACL entries support "LOG-SYSLOG", e.g.,
                        - openconfig-acl:sequence-id: 10
                          openconfig-acl:actions:
                            openconfig-acl:config:
                              openconfig-acl:forwarding-action: 'REJECT'
                              openconfig-acl:log-action: 'LOG_SYSLOG'
We our implementation converts 'LOG_SYSLOG' to log-input. I'll change the XR NED to OC to translate line "deny ipv4 any any log" to "deny ipv4 any any log-input". It's one or the other (log or log-input) with OpenConfig.

Is log-input OK?

**Comment by stmosher** [comment]
Fixed in PR #133
